policies/rudder-cli/src/logs.rs

```diff
@@ -52,111 +50,85 @@ impl FromStr for OutputFormat {
     }
 }
 
+/// `file_log` is a `(log_dir, file_name)` tuple
 pub fn init(
     verbose: u8,
     quiet: bool,
     format: OutputFormat,
-    log_file: Option<&str>,
-) -> Option<tracing_appender::non_blocking::WorkerGuard> {
+    file_log: Option<(&Path, &str)>,
+) -> Result<()> {
     let level = match (verbose, quiet) {
         (0, true) => LevelFilter::WARN,
         (0, false) => LevelFilter::INFO,
         (1, _) => LevelFilter::DEBUG,
         (_, _) => LevelFilter::TRACE,
     };
-    let global_filter = EnvFilter::builder()
-        .from_env_lossy()
-        .add_directive(level.into());
-
     // Already handled by the colored crate by default for other output
     let no_color = env::var("NO_COLOR").is_ok();
-    let console_layer = if format == OutputFormat::Human {
-        let human_log = tracing_subscriber::fmt::layer()
-            .without_time()
-            .with_target(false)
-            .with_writer(std::io::stderr)
-            .event_format(fmt::format().compact().without_time().with_ansi(!no_color));
-        Some(human_log)
-    } else {
-        None
-    };
 
-    let json_layer = if format == OutputFormat::Json {
-        let json_log = tracing_subscriber::fmt::layer()
-            .without_time()
-            .with_target(false)
-            .event_format(fmt::format().json());
-        Some(json_log)
+    // Formatters
+    let human_fmt = fmt::format()
+        .compact()
+        .without_time()
+        .with_target(false)
+        .with_ansi(!no_color);
+    let json_fmt = fmt::format().without_time().with_target(false).json();
+
+    // Layers
+    let human = tracing_subscriber::fmt::layer()
+        .event_format(human_fmt.clone())
+        .with_writer(io::stderr)
+        .with_filter(
+            EnvFilter::builder()
+                .from_env_lossy()
+                .add_directive(level.into()),
+        );
+
+    let file = if let Some((log_dir, log_file_prefix)) = file_log {
+        let file_writer = RollingFileAppender::builder()
+            .filename_prefix(log_file_prefix)
+            // .log extension
+            .filename_suffix("log")
+            .build(log_dir)
+            .context(format!(
+                "Failed to initialize rolling log file in '{}'",
+                log_dir.display()
+            ))?;
+        Some(
+            tracing_subscriber::fmt::layer()
+                .event_format(human_fmt)
```

Comment:
Shouldn't we add a timestamp for the logged lines in the file ?

Reply:
_Bien vu_